test/features/currency/currency_dropdown_test.dart

```diff
@@ -0,0 +1,79 @@
+import 'package:didpay/features/currency/currency_dropdown.dart';
+import 'package:didpay/features/payment/payment_amount_state.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+  
+import '../../helpers/test_data.dart';
+import '../../helpers/widget_helpers.dart';
+
+void main() async {
+  await TestData.initializeDids();
+
+  late ValueNotifier<PaymentAmountState?> mockState;
+  const testCurrency = 'USD';
+
+  setUp(() {
+    // Initialize mockState with a non-null offeringsMap
+    mockState = ValueNotifier<PaymentAmountState?>(
+      PaymentAmountState(
+        offeringsMap: TestData.getOfferingsMap(),
+      ), // Use TestData as suggested
+    );
+  });
+
+  Widget createWidgetUnderTest() {
+    return WidgetHelpers.testableWidget(
+      // Use WidgetHelpers for consistent test setup
+      child: CurrencyDropdown(
+        paymentCurrency: testCurrency,
+        state: mockState,
+      ),
+    );
+  }
+
+  testWidgets('CurrencyDropdown shows the correct initial currency',
+      (tester) async {
+    await tester.pumpWidget(createWidgetUnderTest());
+
+    // Check if the currency label displays the initial currency
+    expect(find.text(testCurrency), findsOneWidget);
+  });
+
+  testWidgets('CurrencyDropdown opens the Select Currency modal when tapped',
+      (tester) async {
+    await tester.pumpWidget(createWidgetUnderTest());
+
+    // Tap on the currency dropdown button
+    await tester.tap(find.text(testCurrency));
+    await tester.pumpAndSettle(); // Wait for the modal to open
+
+    // Verify if the modal is displayed
+    expect(
+      find.text('Select currency'),
+      findsOneWidget,
+    ); // Look for modal text or widget
+  });
+
+  testWidgets('CurrencyDropdown updates state when a currency is selected',
+      (tester) async {
+    await tester.pumpWidget(createWidgetUnderTest());
+
+    // Open the modal by tapping on the currency dropdown button
+    await tester.tap(
+      find.text(testCurrency),
+    ); // Using the 'USD' label text to find the button
+    await tester.pumpAndSettle();
+
+    // Simulate selecting a currency by updating the mockState directly
+    const selectedCurrency = 'EUR';
+    mockState.value = PaymentAmountState(
+      filterCurrency:
+          selectedCurrency, // Assuming 'filterCurrency' is the correct field for currency selection
+    );
```

Comment:
instead of updating `mockState` directly, we should use `tester.tap()` to select a new currency. pull in the [latest changes on main](https://github.com/TBD54566975/didpay/commit/89bbc7116f5aa5559e5cef69ae7c446eb60acc6c) to use an updated version of `TestData` to make this easier for you to do.